src/main/java/com/google/sps/servlets/PotentialMatchesServlet.java

```diff
@@ -60,46 +70,92 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   /**
-  * Loads the next potential match for a user
+  * Loads the match information for a specified user if match information is not
+  * already in datastore.
   *
-  * <p>Currently is just loading all the possible matches and obtaining one of them.
+  * <p>Creates new match information entity for user if it had not already been stored.
+  *
+  * @param userID The ID of the user who's match information is being checked/loaded into datastore
+  * @return The datastore entity of the user's match information
+  */
+  private Entity loadUserMatchInformation(String userID) {
+    Entity matchInfoEntity = datastore.prepare(new Query(MATCH_INFO_ENTITY).setFilter(
+      new FilterPredicate(USER_ID_PROPERTY, FilterOperator.EQUAL, userID))).asSingleEntity();
+    
+    if (matchInfoEntity == null) {
+      return addMatchInfoToDatastore(userID);
+    }
+    return matchInfoEntity;
```

Comment:
supernit: this would be ever-so-slightly easier to read as

```java
if (matchInfoEntity != null) {
  return matchInfoEntity;
}
return addMatchInfoToDatastore(userId);
```

(acts on `matchInfoEntity` immediately after the null check, so that the reader doesn't need to think about that variable anymore when reading forward)

Reply:
Got it!